### PR TITLE
(test) O3-3610 : Unit test for rendering tutorials

### PR DESCRIPTION
**Pull request by Vijaykv5**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.

## Summary

Added test for rendering tutorials.

## Other

[O3-3610](https://openmrs.atlassian.net/browse/O3-3610)

[O3-3610]: https://openmrs.atlassian.net/browse/O3-3610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tutorial/modal.component.test.tsx [hunk 1]
```diff
@@ -41,16 +54,33 @@ describe('TutorialModal', () => {
...
+    await user.click(walkthroughButton);
 
+    await waitFor(() => {
+      mockTutorialData[0].steps.forEach((step) => {
+        const element = screen.queryByText(step.content);
+        if (element) {
+          console.log('Found element:', element);
+        } else {
+          console.log('Element not found for:', step.content);
+        }
```
Comment (Piumal1999) [inline src/tutorial/modal.component.test.tsx:81]: No need to use console logs here. You can use an assertion.

Comment (jayasanka-sack) [reply src/tutorial/modal.component.test.tsx:81]: @Vijaykv5

File: src/tutorial/modal.component.test.tsx [hunk 2]
```diff
@@ -41,16 +54,33 @@ describe('TutorialModal', () => {
...
 
-  test.todo('renders tutorials properly');
+  test('renders tutorials properly', async () => {
+    const user = userEvent.setup();
+    render(<TutorialModal open={true} onClose={jest.fn()} />);
+
+    const walkthroughButton = screen.getByText('Basic Tutorial').closest('.tutorialItem').querySelector('.walkthrough');
+    await user.click(walkthroughButton);
 
+    await waitFor(() => {
```
Comment (jayasanka-sack) [inline src/tutorial/modal.component.test.tsx:74]: What this test should do is it should check if it lists down all the tutorials. No need to click them nor to go through the steps. 

Should be something like:

```
for all tutorial in tutorials:
    expect the tutorial title and description is loaded properly
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]

**Review by jayasanka-sack — APPROVED** [review]
Thanks @Vijaykv5 !

**Pull request merged**